### PR TITLE
Remove unused headers in servers

### PR DESCRIPTION
**Pull request by YYF233333**
Followup #100564 

Cleanup includings in `servers/`, remove unused headers, replace headers used for transitive include, move headers down in the include chain if possible.

### CODE REVIEW

**Comment by akien-mga** [comment]
Seems like `modules/camera/camera_feed_linux.cpp` needs to include RenderingServer, and for some reason there's a new warning in an iOS file triggered by these changes (likely a pre-existing warning that just happened to be raised now because some header changes caused recompilation of the cached object).

```
platform/ios/keyboard_input_view.mm:187:27:{187:25-187:26}{187:29-187:87}: error: comparison of integers of different signs: 'NSInteger' (aka 'long') and 'unsigned long' [-Werror,-Wsign-compare,2]
                for (NSInteger i = 0; i < MIN([substringToDelete length], [substringToEnter length]); i++) {
                                       ~ ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

File: platform/ios/keyboard_input_view.mm [hunk 1]
```diff
@@ -184,7 +184,7 @@ - (void)observeTextChange:(NSNotification *)notification {
 
 	NSInteger skip = 0;
 	if (substringToDelete != nil) {
-		for (NSInteger i = 0; i < MIN([substringToDelete length], [substringToEnter length]); i++) {
+		for (NSInteger i = 0; i < (NSInteger)MIN([substringToDelete length], [substringToEnter length]); i++) {
```
Comment (bruvzg) [inline platform/ios/keyboard_input_view.mm:187]:
```suggestion
		for (NSUInteger i = 0; i < MIN([substringToDelete length], [substringToEnter length]); i++) {
```

**Comment by akien-mga** [comment]
Looks good! Could you squash the commits? See [PR workflow](https://docs.godotengine.org/en/latest/contributing/workflow/pr_workflow.html#the-interactive-rebase) for instructions.

**Comment by YYF233333** [comment]
Sorry, left for a while. Squashed.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
BTW, I'd be interested in knowing your workflow for identifying these unused headers.

I've tried to use iwyu for this exact use case in the past, but I could only get it to *add* includes that are used through existing explicit includes (which would bloat things considerably, as we tend to expect that including e.g. `node_2d.h` is sufficient and you don't need also explicit `canvas_item.h`, `node.h`, `object.h`, etc.). But indeed a "don't-include-what-you-don't-use" is sorely needed for C/C++ codebases.

**Comment by YYF233333** [comment]
> BTW, I'd be interested in knowing your workflow for identifying these unused headers.

Actually clangd is enough for identifying unused header. Everytime I open a file in vscode there are unused header warnings which is annoying and finally I decide to clean up them.

iwyu gives some more aggressive suggestions, like replace some headers with forward declarations.
e.g.
```bash
core/os/memory.h should add these lines:
#include "core/typedefs.h"            // for _FORCE_INLINE_, _ALWAYS_INLINE_
template <typename T> class SafeNumeric;

core/os/memory.h should remove these lines:
- #include <type_traits>  // lines 39-39
- #include "core/templates/safe_refcount.h"  // lines 35-35
```
I'm currently evaluating how much build time they can save, and may pick some big improvements (if there are any) without sacrificing readability.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**